### ISSUE TITLE
Memory testing on Ubuntu 22.04 LTS

### ISSUE THREAD

**Issue by fcooper8472** [Chaste/Chaste#87]
The new CI workflow for memory testing runs on a self-hosted runner (currently the new Chaste server in Oxford).  This is necessary because the full test suite takes slightly longer than the allowed 6 hours on a GitHub hosted runner.

The server is running Ubuntu 22.04 LTS, which supplies Valgrind-3.18.1.

An example set out outputs from this version is available here:
https://chaste.github.io/memory-testing/2023-03-16_14-00-34/index.html

Every test is marked as leaking, and this may be related to the following warning:
```
==923004== WARNING: pwritev(vector[...]) is an obsolete suppression line not supported in valgrind 3.15 or later.
==923004== You should replace [...] by a specific index such as [0] or [1] or [2] or similar
```

It may be the case that our memory testing suppressions are out of date and need re-creating.

Previous memory testing, available for the time being on the BuildBot waterfall, does not have the same issue, and the overall memory test suite is passing:
https://chaste.cs.ox.ac.uk/buildbot/Nightly%20MemTest/memtest1804/

**Comment by mirams** [comment]
Almost all of them fail, 15 also have that warning and are passing (search https://chaste.github.io/memory-testing/2023-03-16_14-00-34/index.html for "No leaks"):
e.g. 
 * https://chaste.github.io/memory-testing/2023-03-16_14-00-34/TestHeartRegionCodes_valgrind.txt
 * https://chaste.github.io/memory-testing/2023-03-16_14-00-34/TestAirwayBranch_valgrind.txt
 * https://chaste.github.io/memory-testing/2023-03-16_14-00-34/TestPetscTools2_valgrind.txt 
 * https://chaste.github.io/memory-testing/2023-03-16_14-00-34/TestImpedancePostProcessor_valgrind.txt
 * https://chaste.github.io/memory-testing/2023-03-16_14-00-34/TestWritingTestsTutorial_valgrind.txt
seems like these are readers, writers, postprocessors and stuff, perhaps things not using much of core Chaste as if there is something fundamental in `global` causing the problem - ideas @jmpf ? Not ever calling the main PetscInitialise method?

**Comment by jmpf** [comment]
After a few failed attempts to get a predictable suppression file with <100 suppressions I went to a back-to-basics approach.  There are so many OpenMPI leaks from PetscInitalize/PetscFinalize that I temporarily dropped Chaste in favour of "Hello world".  The code
```#include "petsc.h"
int main( int argc, char *argv[]) {
	PetscInitialize(&argc,&argv,PETSC_NULL,PETSC_NULL);
	//PetscPrintf(PETSC_COMM_WORLD,"Hello World\n");
	printf("Hello\n");
	PetscFinalize();

	return 0;
}
```
tested with `valgrind --leak-check=full --suppressions=hello.supp  --gen-suppressions=all ./a.out` required more than 30 suppressions.  By repeated testing I edited it down to [hello.supp.txt](https://github.com/Chaste/Chaste/files/11404569/hello.supp.txt) with 11 wildcard style suppressions.

This test was done on Ubuntu 22.04 with OpenMPI 4.1.2-2ubuntu1 and PETSc 3.15.5+dfsg1-1build2
